src/runtime.js

```diff
@@ -173,7 +148,18 @@ function adoptShared(global, callback) {
   global.AMP.toggleExperiment = toggleExperiment.bind(null, global);
 
   // Run specific setup for a single-doc or shadow-doc mode.
-  callback(global);
+  callback(global, modules);
+
+  /**
+   * @param {function(!Object)|{n:string, f:function(!Object)}} fnOrStruct
+   */
+  function installModule(fnOrStruct) {
+    if (typeof fnOrStruct == 'function') {
+      fnOrStruct(global.AMP);
```

Comment:
So, this is backwards compatible for modules as well? During roll out you need to expect extensions to use the old syntax.